### DISCUSSION TITLE
is numer db checker

### DISCUSSION THREAD

**Discussion by Urbaninja25** [supabase/supabase#20347] [Questions]
hello,

i am new to proggraming and thats why i am using supabase.

i fail to create function which will check if users price value they try to add in db is only numeric 

for that i use these function which i wanna use as trigger afterwards in the middle of user initiated  insert and actually store it in db  :

CREATE OR REPLACE FUNCTION public.price_is_numeric_check()
  RETURNS trigger AS $$
BEGIN
  IF NEW.value ~ '^[0-9]+$' THEN
    RETURN NEW;
  ELSE
    NEW.value := substring(NEW.value FROM '^[0-9]+');
    RETURN NEW;
  END IF;
END;
$$ LANGUAGE plpgsql;

supabase just return err massage saying i have syntaxt err 

if someone can provide correct function i will be pleased! i just need to see it once and then will do my own,i promise ! 

thanks in advance ~

**Comment by GaryAustin1** [comment]
Are you running this code in the SQL editor or in the Function UI?  If in the UI you only enter the code starting with begin and ending with end;

**Reply by Urbaninja25** [reply]
hello @GaryAustin1  firstly thank u for advising  me that !

ye i have read that and i dont have any syntax problem anymore but i cant make my function run .

i have price field(data type = int) and i wanna create trigger (which i did) which will execute my function before insert&update .

main purpuse of my function is to run on every raw modification and exclude non numeric values from price field before store it .

my function looks like that :
BEGIN
  NEW.price := REGEXP_REPLACE(NEW.price, '[^0-9]', '', 'g'); -- Remove non-numeric characters
  NEW.price := NEW.price::int;  -- Convert to integer
  RETURN NEW;
END;

**Reply by Urbaninja25** [reply]
ps: whenever  i try to insert something like 4,500 i get err,-"non intiger value"
